### ISSUE TITLE
[PROPOSAL] Adding Mines as buildables.

### ISSUE THREAD

**Issue by Quartofel** [colobot/colobot#885]
Mines could add aditional possibilities in Code Battles. They could be decent way to deny area from land attacks. They could be built from Titanium with _build(Mine);_ command.

**Comment by tomangelo2** [comment]
According to lore, mines are deprecated technology due to being dangerous for mission, but some cheap close-range defense system would be handy in code battles. I would think if they would need some changes in this mode, currently they seems for me a bit useless, as you might want to build DefenseTower, that could cover area of many mines.

**Comment by tomaszkax86** [comment]
Mines were considered dangerous for main missions in the game but Code Battles are not part of main missions. I think we should add mines as build-able structures to non-mission game modes, and potentially many other types of buildings, like barriers, starting and ending platforms, and others. I like the idea of using mines as defense in bots vs aliens type of battles. If one player can build mines, the other player will have to write code to avoid them.

**Comment by Quartofel** [comment]
@tomangelo2 I agree that mines are not-so-useful in comparison with DefenseTowers, but the main power of Code Battles might be diversity.
It's not hard to disable, let's sayy, Winged bots and DefenseTowers for the sake of changed gameplay.
Additionally, when this game will be put on the Steam, players will appear, and their imagination might be really surprising. The more possibilities they'll get, the more amazing things will appear on Workshop.

**Comment by Quartofel** [comment]
And moreover, even regarding lore it makes sense to make Mines as buildables. NASA got the mines technology, they were actually built. It's prohibited when Colobot campaign is going on, but they were used earlier, and Code Battles might be a retrospection - training for the First Expedition for instance.

Well, there could be even a minelayer bot, but I think it's not so important issue to bother Dev team with implementing new bot. Well, maybe in later stage of development...

**Comment by rbcat** [comment]
Well, then, if it's gonna be implemented someday, why constrain it to code battles? Don't forget about user levels too.

I've got another idea into my mind. See, why these mines are so red, so bright? It have always looked strange for me, because mines should be almost unnoticeable... And then I realized that it might be that they look this way because they are supposed to *attract* attention of insects. And for such, you could just write AI for insects so that they behave this way and attack / approach the mines and, of course, explode on them. In such case mines could be actually useful weapon indeed, especially giving that they could be more available, as a cheap and simple weapon when you don't have more power cells around nor plans for defense towers being researched nor a place to build one.
